src/modules/tools/reporter/Reporter.cpp

```diff
@@ -0,0 +1,181 @@
+      You should have received a copy of the GNU General Public License along with Smoothie. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "libs/Module.h"
+#include "libs/Kernel.h"
+#include "Reporter.h"
+#include "Gcode.h"
+#include "StreamOutput.h"
+#include "TemperatureControlPublicAccess.h"
+#include "TemperatureControl.h"
+#include "PlayerPublicAccess.h"
+#include "Player.h"
+#include "ExtruderPublicAccess.h"
+#include "PublicData.h"
+#include "Robot.h"
+#include "ZProbe.h"
+#include "modules/robot/Conveyor.h"
+
+#include "version.h"
+#include "checksumm.h"
+#include "ConfigValue.h"
+#include "Config.h"
+
+#include <math.h>
+#include <vector>
+#include <algorithm>
+
+#define reporter_checksum          CHECKSUM("reporter")
+#define enable_checksum            CHECKSUM("enable")
+#define probe_height_checksum      CHECKSUM("probe_height")
+
+
+bool less_than_padtemp(const pad_temperature& leftSide, const pad_temperature& rightSide)
+{
+        if(leftSide.designator == "B" && rightSide.designator == "T")
+        {
+                return true;
+        }else if(leftSide.designator == rightSide.designator) {
+                return (leftSide.id < rightSide.id);
+        }
+
+        return false;
+};
+
+Reporter::Reporter(){
+}
+
+Reporter::~Reporter(){
+}
+
+void Reporter::on_module_loaded(){
+        // Exit if this module is not enabled
+        if ( !THEKERNEL->config->value( reporter_checksum, enable_checksum )->by_default(false)->as_bool() ) {
+                delete this;
+                return;
+        }
+
+        this->register_for_event(ON_GCODE_RECEIVED);
+}
+
+void Reporter::on_gcode_received(void *argument){
+        Gcode *gcode = static_cast<Gcode *>(argument);
+        if (gcode->has_m && gcode->m == 408 && gcode->has_letter('S')) {
+
+                switch(static_cast<int>(gcode->get_value('S'))) {
+                case 0:
+                        gcode->stream->printf("Getting M408 S0 status");
+                        // Beginning
+                        // Find the status
+                        //I=idle, P=printing from SD card, S=stopped (i.e. needs a reset),
+                        //C=running config file, A=paused, D=pausing, R=resuming, B=busy (running a macro)
+                        gcode->stream->printf("{\"status\":\"");
+                        if(THEKERNEL->conveyor->is_queue_empty()) {
+                                gcode->stream->printf("I");
+                        }else{
+                                gcode->stream->printf("P");
+                        }
+                        if (THEKERNEL->is_halted()) {
+                                gcode->stream->printf("S");
+                        }
+                        void *returned_data;
+                        if (PublicData::get_value( player_checksum, is_suspended_checksum, &returned_data )) {
```

Comment:
This would result in "PA" if queue not empty and paused? Suggest to make code a little safer, so only "I", "P", "S", or "A" are returned.